docs/source/index.rst

```diff
@@ -1,20 +1,204 @@
-.. GraphQL Compiler documentation master file, created by
-   sphinx-quickstart on Mon Jul 22 00:42:04 2019.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+The GraphQL Compiler
+====================
 
-Welcome to GraphQL Compiler's documentation!
-============================================
+.. EDUCATIONAL: The toctree (Table Of Contents Tree) specifies the table of contents at the left
+   side of the page. The maxdepth indicates which headers to include in the table of contents.
+   :hidden: specifies not to include a table of contents in this page.
+
+.. ISSUE: Tree doesn't include subsections for `self` in toctree.
+          https://github.com/sphinx-doc/sphinx/issues/2103
 
 .. toctree::
    :maxdepth: 2
-   :caption: Contents:
+   :hidden:
+
+   Home <self>
+
+.. EDUCATIONAL: The meta field below is for search engine optimization.
+.. meta::
+   :description: Turn complex GraphQL queries into optimized database queries.
+   :keywords: graphql compiler, database, orientdb, sql
+
+The GraphQL Compiler is a library that simplifies database querying and exploration by exposing one
+common query language for multiple database backends.  The query language is:
+
+.. EDUCATIONAL: The pattern below is what you would call a definition list in restructuredtext.
+   The "terms" get special rendering in the readthedocs html file.
+
+Written in valid GraphQL syntax
```

Comment:
I'm not sure about the formatting here. I think this is more typically used for function definitions. e.g. https://pika.readthedocs.io/en/stable/modules/connection.html

I would consider making this a bulleted list or a few small headers.

Reply:
Note how pika also points to the source code in the docs - this would be an awesome feature to have in our docs.

Reply:
I was going for the way this looks https://docs.readthedocs.io/en/stable/index.html
This is also a definition list though for some reason it is rendered differently. I made a github issues asking why.  

As to sphinx autodoc, I think it is a good idea, but it's not relevant here.

Reply:
I changed the definition list terms for bolded text instead. I will try to fix this in the future to look as to how I signaled it above. For now, I would prefer to move on.

Reply:
Actually, for now, I will just add a TODO. I will fix this once they get back to me.

Reply:
Ok they responded and pointed out that the sphinx read the docs theme is not ready for Sphinx 2.0. I bumped down the version of Sphinx to 1.8 which is the version they are using.